### PR TITLE
Cleanup exact flow types

### LINKED ISSUE #90: Cleanup exact Flow types
This essentially means to get rid of `{| |}` syntax. Preferably via Eslint rule, so it's enforced and auto-fixable.

### PR DESCRIPTION
**Pull request by tbergq**
closes #90

### CODE REVIEW

**Comment by mrtnzlml** [comment]
@tbergq Out of curiosity: did you use any tool for this or simple find+replace?

**Comment by tbergq** [comment]
> @tbergq Out of curiosity: did you use any tool for this or simple find+replace?

Just VS-code search and replace

File: .flowconfig [hunk 1]
```diff
@@ -29,7 +29,6 @@
 ; https://flow.org/en/docs/config/lints/
 [lints]
 all=error
-ambiguous-object-type=off
```
Comment (mrtnzlml) [inline .flowconfig]: Is this ON by default now?

File: src/example-relay/src/Homepage/__generated__/HomepageQuery.graphql.js [hunk 2]
```diff
@@ -10,16 +10,16 @@ import type { ConcreteRequest } from 'relay-runtime';
 type LocationsPaginatedBidirectional_data$ref = any;
 type LocationsPaginatedRefetch_data$ref = any;
 type LocationsPaginated_data$ref = any;
-export type HomepageQueryVariables = {|
```
Comment (mrtnzlml) [inline src/example-relay/src/Homepage/__generated__/HomepageQuery.graphql.js]: 🤔 We should not change this file manually. Or any other generated file.

File: .flowconfig (hunk 1 above)
Comment (tbergq) [reply .flowconfig]: I am a little bit confused here, I had to remove it from strict, but I can see it gives no errors when it is off, and no errrors when on. 

On the strict config I got errors 🤔

Comment (tbergq) [reply .flowconfig]: It says `Like implicit-inexact-object, except triggers even when the exact_by_default option is set to true.`

Which seems like it does not, so, I can leave it to off

File: src/example-relay/src/Homepage/__generated__/HomepageQuery.graphql.js (hunk 2 above)
Comment (tbergq) [reply src/example-relay/src/Homepage/__generated__/HomepageQuery.graphql.js]: Good point, I should have left these filtered out.

Comment (tbergq) [reply src/example-relay/src/Homepage/__generated__/HomepageQuery.graphql.js]: Should be back to normal now

Comment (mrtnzlml) [reply src/example-relay/src/Homepage/__generated__/HomepageQuery.graphql.js]: I can still see some __generated__ files modified in this PR. For example: `src/example-relay/src/LocalForm/__generated__/LocalFormQuery.graphql.js`

Comment (tbergq) [reply src/example-relay/src/Homepage/__generated__/HomepageQuery.graphql.js]: Sorry, I thought just running the relay-compiler was enough. They should be gone now.

**Comment by mrtnzlml** [comment]
@tbergq Do we have some Eslint rule which would prevent us from using this syntax?

**Comment by tbergq** [comment]
No, I don’t think there is, using the | is still valid syntax, just not necessary

**Comment by michalsanger** [comment]
The rule would be handy. You could use the `|` syntax only when you explicitly ignore that rule.

**Comment by tbergq** [comment]
Not really sure, I am even starting to doubt that we should merge this one. 

Since `{||}` always means exact, by removing it, we are also assuming that our consumers are using exact by default, which they are maybe not doing.

While keeping the | syntax, it will be exact no matter what flowconfig the consumer is using

**Comment by mrtnzlml** [comment]
@tbergq You mean it would not be compatible with the other repositories which are not yet switched? Since we are providing the NPM libraries? So we should stick with the `{| |}` syntax for as long as possible until it's a real problem?

That sounds like a very valid and smart argument why to close this PR, yes. We are currently enforcing the `{| |}` syntax, correct?

**Comment by tbergq** [comment]
> You mean it would not be compatible with the other repositories which are not yet switched?

Exactly yes. 

> We are currently enforcing the {| |} syntax, correct?

Yes, we are, through the require explicit inexact eslint rule.

So I am closing this PR 😊

**Pull request closed without merging**